app/Http/Controllers/API/v1/Auth/ForgotPasswordController.php

```diff
@@ -0,0 +1,144 @@
+        ]);
+
+        $email = $request->email;
+        $verificationCode = random_int(100000, 999999);
+        $expiresAt = now()->addMinutes(15);
+
+        VerificationCode::updateOrCreate(
+            ['contact' => $email, 'purpose' => 'password_reset'],
+            ['code' => $verificationCode, 'expires_at' => $expiresAt]
+        );
+
+        $user = User::where('email', $email)->first();
+        $this->notificationService->sendEmailNotification([
+            'to' => $email,
+            'subject' => 'Password Reset Code',
+            'body' => "Your password reset code for Trakli is: $verificationCode",
+        ]);
+
+        return $this->success([], 'Password reset code sent successfully.');
+    }
+
+    #[OA\Post(
+        path: '/password/verify-code',
+        tags: ['Authentication'],
+        summary: 'Verify password reset code',
+        requestBody: new OA\RequestBody(
+            required: true,
+            content: new OA\JsonContent(
+                required: ['email', 'code'],
+                properties: [
+                    new OA\Property(
+                        property: 'email',
+                        type: 'string',
+                        format: 'email',
+                        description: 'The email address of the user'
+                    ),
+                    new OA\Property(
+                        property: 'code',
+                        type: 'integer',
+                        description: 'The verification code sent to the user'
+                    ),
+                ]
+            )
+        ),
+        responses: [
+            new OA\Response(
+                response: 200,
+                description: 'Verification successful',
+                content: new OA\JsonContent(
+                    properties: [
+                        new OA\Property(
+                            property: 'message',
+                            type: 'string',
+                            example: 'Code verified successfully.'
+                        ),
+                    ]
+                )
+            ),
+            new OA\Response(
+                response: 422,
+                description: 'Validation error'
+            ),
+            new OA\Response(
+                response: 400,
+                description: 'Invalid or expired code'
+            ),
+        ]
+    )]
+    public function verifyPasswordResetCode(Request $request)
+    {
+        $request->validate([
+            'email' => 'required|email|exists:users,email',
+            'code' => 'required|integer',
+        ]);
+
+        $codeEntry = VerificationCode::where('contact', $request->email)
+            ->where('purpose', 'password_reset')
+            ->first();
+
+        if (! $codeEntry || $codeEntry->code != $request->code || $codeEntry->isExpired()) {
+            return $this->failure('Invalid or expired code.', 400);
+        }
+
+        return $this->success([], 'Code verified successfully.');
```

Comment:
Verification code is retrieved but no action is performed on the code to change the status or the user to change verification status.